pkg/vault/test/foundry/utils/BaseVaultTest.sol

```diff
@@ -131,7 +131,13 @@ abstract contract BaseVaultTest is VaultStorage, BaseTest, Permit2Helpers {
 
     function setUp() public virtual override {
         BaseTest.setUp();
+        _setUpBaseVaultTest();
+        // Add initial liquidity
+        initPool();
+        vaultConvertFactor = vault.getConvertFactor();
+    }
 
+    function _setUpBaseVaultTest() internal {
```

Comment:
Is there any particular reason for this refactor?

Reply:
> Later on it'd be good to have a test that checks all of this against different pool types with different configurations. The base vault test architecture creates a single pool and runs the test, but that's not very representative. We can do that in a later stage.

This may help to implement this point